core/src/main/java/org/apache/iceberg/SnapshotProducer.java

```diff
@@ -226,7 +226,9 @@ public Snapshot apply() {
             manifestList,
             snapshotId(),
             parentSnapshotId,
-            sequenceNumber)) {
+            sequenceNumber,
+            ops.current().properties().get(TableProperties.AVRO_COMPRESSION),
+            ops.current().propertyAsNullableInt(TableProperties.AVRO_COMPRESSION_LEVEL))) {
```

Comment:
This shouldn't refer to `ops`. Instead, use `base`. We want to avoid referencing `ops` directly even if it is available.

Reply:
Ok. The call here to `Manifests.write` uses `ops.current().formatVersion()` for the format version; should I change that too? There are other uses of `ops.current()` in `SnapshotProducer` as well; should I change those too?

Reply:
I changed `ops.current()` to `base` in the methods that I modified, but not in the other methods.